Mathlib/CategoryTheory/WithTerminal.lean

```diff
@@ -46,79 +46,71 @@ variable (C : Type u) [Category.{v} C]
-Case conversion may be inaccurate. Consider using '#align category_theory.with_terminal.hom CategoryTheory.WithTerminal.Homₓ'. -/
 /-- Morphisms for `with_terminal C`. -/
-@[simp, nolint has_nonempty_instance]
+-- porting note: unsupported `nolint has_nonempty_instance`
+@[simp]
 def Hom : WithTerminal C → WithTerminal C → Type v
   | of X, of Y => X ⟶ Y
-  | star, of X => PEmpty
+  | star, of _ => PEmpty
   | _, star => PUnit
 #align category_theory.with_terminal.hom CategoryTheory.WithTerminal.Hom
 
-/- warning: category_theory.with_terminal.id -> CategoryTheory.WithTerminal.id is a dubious translation:
-lean 3 declaration is
-  forall {C : Type.{u2}} [_inst_1 : CategoryTheory.Category.{u1, u2} C] (X : CategoryTheory.WithTerminal.{u1, u2} C _inst_1), CategoryTheory.WithTerminal.Hom.{u1, u2} C _inst_1 X X
-but is expected to have type
-  PUnit.{max (succ (succ u1)) (succ (succ u2))}
-Case conversion may be inaccurate. Consider using '#align category_theory.with_terminal.id CategoryTheory.WithTerminal.idₓ'. -/
 /-- Identity morphisms for `with_terminal C`. -/
 @[simp]
 def id : ∀ X : WithTerminal C, Hom X X
-  | of X => 𝟙 _
+  | of _ => 𝟙 _
   | star => PUnit.unit
 #align category_theory.with_terminal.id CategoryTheory.WithTerminal.id
 
-/- warning: category_theory.with_terminal.comp -> CategoryTheory.WithTerminal.comp is a dubious translation:
-lean 3 declaration is
-  forall {C : Type.{u2}} [_inst_1 : CategoryTheory.Category.{u1, u2} C] {X : CategoryTheory.WithTerminal.{u1, u2} C _inst_1} {Y : CategoryTheory.WithTerminal.{u1, u2} C _inst_1} {Z : CategoryTheory.WithTerminal.{u1, u2} C _inst_1}, (CategoryTheory.WithTerminal.Hom.{u1, u2} C _inst_1 X Y) -> (CategoryTheory.WithTerminal.Hom.{u1, u2} C _inst_1 Y Z) -> (CategoryTheory.WithTerminal.Hom.{u1, u2} C _inst_1 X Z)
-but is expected to have type
-  PUnit.{max (succ (succ u1)) (succ (succ u2))}
-Case conversion may be inaccurate. Consider using '#align category_theory.with_terminal.comp CategoryTheory.WithTerminal.compₓ'. -/
 /-- Composition of morphisms for `with_terminal C`. -/
 @[simp]
 def comp : ∀ {X Y Z : WithTerminal C}, Hom X Y → Hom Y Z → Hom X Z
-  | of X, of Y, of Z => fun f g => f ≫ g
-  | of X, _, star => fun f g => PUnit.unit
-  | star, of X, _ => fun f g => PEmpty.elim f
-  | _, star, of Y => fun f g => PEmpty.elim g
+  | of _X, of _Y, of _Z => fun f g => f ≫ g
+  | of _X, _, star => fun _f _g => PUnit.unit
+  | star, of _X, _ => fun f _g => PEmpty.elim f
+  | _, star, of _Y => fun _f g => PEmpty.elim g
   | star, star, star => fun _ _ => PUnit.unit
 #align category_theory.with_terminal.comp CategoryTheory.WithTerminal.comp
 
 instance : Category.{v} (WithTerminal C) where
   Hom X Y := Hom X Y
   id X := id _
-  comp X Y Z f g := comp f g
+  comp := comp
+  -- Porting note : need to do cases analysis
+  comp_id {X Y} _ := by aesop
+  id_comp {X Y} _ := by aesop
```

Comment:
Do I understand correctly that these can not be deleted, because `aesop` can now do cases, but `aesop_cat` doesn't pick up that power?

Reply:
No, they can just be deleted. I've done this now.